### PR TITLE
DB file handler script thingy

### PR DESCRIPTION
**Pull request by nbitzz**
This is fucking terrifying please do not ask why it's so complex

### CODE REVIEW

**Comment by Jack5079** [comment]
I have no idea

File: src/server/lib/dbfile.ts [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+// this is fucking stupid why did i write this
+
+class Activity {
+    
+    _write: () => Promise<any>
+    destroy: () => void
+
+    goal: number = Date.now()
+    lastWrt: number = Date.now()
```
Comment (Jack5079) [inline src/server/lib/dbfile.ts:17]: last huh? Did you mean `lastWrite`? Name it that dumbass
```suggestion
    lastWrite: number = Date.now()
```

Comment (nbitzz) [reply src/server/lib/dbfile.ts:17]: I'm gonna fucking kill myself

File: src/server/lib/dbfile.ts [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    clock? : { type: "precise", id: NodeJS.Timeout } | { type: "tick", id: NodeJS.Timeout, lastGoal: number }
+
+    constructor(writeFunc: () => Promise<any>, destroyFunc: () => void) {
+        this._write = writeFunc
```
Comment (Jack5079) [inline src/server/lib/dbfile.ts:21]: tired

Comment (Jack5079) [reply src/server/lib/dbfile.ts:21]: might not be obvious but referring to the whole `this._write =` shit terrifying

Comment (nbitzz) [reply src/server/lib/dbfile.ts:21]: My brain's off I don't know why I did that

**Comment by nbitzz** [comment]
Actually it's probably good enough

**Pull request merged**